kie-server-parent/kie-server-maven-plugin/src/main/java/org/kie/server/gateway/KieServerGateway.java

```diff
@@ -0,0 +1,112 @@
+package org.kie.server.gateway;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.kie.server.api.marshalling.MarshallingFormat;
+import org.kie.server.api.model.*;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.KieServicesFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by fabio.ercoli@redhat.com on 24/07/2017.
+ */
+public class KieServerGateway {
+
+    public static final Logger LOG = LoggerFactory.getLogger(KieServerGateway.class);
+
+    private final KieServicesConfiguration config;
+    private KieServicesClient client;
+
+    public KieServerGateway(String protocol, String hostname, Integer port, String username, String password, Integer timeout) {
+
+        String serverUrl = new StringBuilder(protocol)
+                .append("://")
+                .append(hostname)
+                .append(":")
+                .append(port)
+                .append("/kie-server/services/rest/server")
+                .toString();
+
+        LOG.info("Server Url {}", serverUrl);
+
+        config = KieServicesFactory.newRestConfiguration(serverUrl, username, password);
+        config.setMarshallingFormat(MarshallingFormat.JSON);
+        config.setTimeout(timeout);
+    }
+
+    public void init() throws MojoFailureException {
+
+        try {
+            client = KieServicesFactory.newKieServicesClient(config);
+        } catch (RuntimeException kieEx) {
+            throw new MojoFailureException("error on enstablish connection with remote server: " + kieEx.getMessage());
+        }
+
+    }
+
+    public void deploy(MavenProject project, String container) throws MojoFailureException {
+
+        KieContainerResource kieContainer = new KieContainerResource();
+        kieContainer.setContainerId(container);
+
+        ReleaseId releaseId = new ReleaseId();
+        releaseId.setGroupId(project.getGroupId());
+        releaseId.setArtifactId(project.getArtifactId());
+        releaseId.setVersion(project.getVersion());
+        kieContainer.setReleaseId(releaseId);
+
+        KieServerConfigItem configItem = new KieServerConfigItem();
+        configItem.setName("RuntimeStrategy");
+        configItem.setValue("PER_PROCESS_INSTANCE");
```

Comment:
this should be configurable as well, default can be set to per process instance but then it should be visible in the logs as well. Would be good to actually print out the details of what is going to be deployed.